### ISSUE TITLE
Throw "Uncaught (in promise) TypeError: headers[headerName].trim is not a function" while using AnthropicBedrock client in Deno

### ISSUE THREAD

**Issue by MMMartt** [anthropics/anthropic-sdk-typescript#350]
While running the example of bedrock client in readme in **Deno**, it throws:
```
error: Uncaught (in promise) TypeError: headers[headerName].trim is not a function
    at getCanonicalHeaders (deno/npm/registry.npmjs.org/@smithy/signature-v4/1.1.0/dist-cjs/getCanonicalHeaders.js:20:62)
    at SignatureV4.signRequest (deno/npm/registry.npmjs.org/@smithy/signature-v4/1.1.0/dist-cjs/SignatureV4.js:120:80)
    at eventLoopTick (ext:core/01_core.js:169:7)
    at async getAuthHeaders (deno/npm/registry.npmjs.org/@anthropic-ai/bedrock-sdk/0.9.1/auth.mjs:50:20)
    at async AnthropicBedrock.prepareRequest (deno/npm/registry.npmjs.org/@anthropic-ai/bedrock-sdk/0.9.1/client.mjs:62:25)
    at async AnthropicBedrock.makeRequest (deno/npm/registry.npmjs.org/@anthropic-ai/sdk/0.19.0/core.mjs:268:9)
```

It seems that bedrock sdk is complaining about headers `'X-Stainless-Runtime-Version'` is not a string. As it is using `headers[headerName].trim()`.

And I found this is set in `core.ts` here:
https://github.com/anthropics/anthropic-sdk-typescript/blob/ad92b0d536508954ee8b6c83e82bca30eefeb298/src/core.ts#L805-L823

`Deno.version` return a object like `{ deno: "1.41.3", v8: "12.3.219.9", typescript: "5.3.3" }` rather than a string, so this error occurs.

It could be fixed easily by converting `Deno.version` to string.

**Comment by RobertCraigie** [comment]
Thanks for reporting! We're looking into this.

**Comment by MMMartt** [comment]
@RobertCraigie When will this be fixed? Currently this sdk is unusable in deno runtime unless edit the local package code manually, this should be easy to fix, should I open a pr?

**Comment by RobertCraigie** [comment]
@MMMartt this fix has been released in `v0.9.2`, have you tried that version?

**Comment by MMMartt** [comment]
Thanks, it's fixed in the new version!
